osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/BundleService.java

```diff
@@ -30,53 +30,54 @@
 import org.opensmartgridplatform.shared.infra.jms.ResponseMessageResultType;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 @Service(value = "domainSmartMeteringBundleService")
-@Transactional(value = "transactionManager")
 public class BundleService {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(BundleService.class);
 
-    @Autowired
-    @Qualifier(value = "domainSmartMeteringOutboundOsgpCoreRequestsMessageSender")
-    private OsgpCoreRequestMessageSender osgpCoreRequestMessageSender;
+    private final OsgpCoreRequestMessageSender osgpCoreRequestMessageSender;
 
-    @Autowired
-    @Qualifier(value = "domainSmartMeteringOutboundWebServiceResponsesMessageSender")
-    private WebServiceResponseMessageSender webServiceResponseMessageSender;
+    private final WebServiceResponseMessageSender webServiceResponseMessageSender;
 
-    @Autowired
-    private DomainHelperService domainHelperService;
+    private final DomainHelperService domainHelperService;
 
-    @Autowired
-    private ActionMapperService actionMapperService;
+    private final ActionMapperService actionMapperService;
 
-    @Autowired
-    private ActionMapperResponseService actionMapperResponseService;
+    private final ActionMapperResponseService actionMapperResponseService;
 
-    @Autowired
-    private MBusGatewayService mBusGatewayService;
+    private final MBusGatewayService mBusGatewayService;
 
-    @Autowired
-    private ManagementService managementService;
+    private final ManagementService managementService;
 
-    @Autowired
-    private ConfigurationMapper configurationMapper;
+    private final ConfigurationMapper configurationMapper;
 
-    @Autowired
-    private FirmwareService firmwareService;
+    private final FirmwareService firmwareService;
 
-    @Autowired
-    private EventService eventService;
+    private final EventService eventService;
 
-    public BundleService() {
-        // Parameterless constructor required for transactions...
+    public BundleService(
+            @Qualifier(value = "domainSmartMeteringOutboundOsgpCoreRequestsMessageSender") OsgpCoreRequestMessageSender osgpCoreRequestMessageSender,
```

Comment:
too many parameters

Reply:
But constructor injection is preferred in Sping!